annotation/core/views.py

```diff
@@ -96,32 +101,52 @@ def annotate(request):
         text = random.choice(group.evaluation_texts.exclude(id__in=seen_set))
     else:
         text = random.choice(EvaluationText.objects.exclude(id__in=seen_set))
-    
+
+    prompt = text.prompt
     sentences = ast.literal_eval(text.body)[:10]
     # remaining = request.session.get('remaining', BATCH_SIZE)
-    
+
+    # Check if the user has a profile object
+    if Profile.objects.filter(user=request.user).exists():
+        is_turker = Profile.objects.get(user=request.user).is_turker
+    else:
+        is_turker = False
+
+    # The %age of all-human examples that will be converted to attention checks for turkers
+    ATTENTION_CHECK_RATE = 0.5
+
+    # Check attention if the user is from Mechanical Turk
+    attention_check = False
+    if is_turker and text.boundary == len(sentences):
+        if random.random() < ATTENTION_CHECK_RATE:
+            prompt.body += " Please choose 'It's all human-written so far.' for every sentence in this example."
+            attention_check = True
+
     print("Here with text_id = {}".format(text.pk))
     return render(request, "annotate.html", {
         # "remaining": remaining,
-        "prompt": text.prompt, 
-        "text_id": text.pk, 
-        "sentences": json.dumps(sentences), 
-        "name": request.user.username, 
+        "prompt": prompt,
+        "text_id": text.pk,
+        "sentences": json.dumps(sentences),
+        "name": request.user.username,
         "max_sentences": len(sentences),
         "boundary": text.boundary,
-        "num_annotations": len(Annotation.objects.filter(annotator=request.user)),
+        "num_annotations": len(Annotation.objects.filter(annotator=request.user, attention_check=False)),
```

Comment:
`len(Annotation.objects.filter(annotator=request.user, attention_check=False))` seems like it could also be a helper function.